atomics/include/desul/atomics/Lock_Based_Fetch_Op.hpp

```diff
@@ -17,7 +17,10 @@ SPDX-License-Identifier: (BSD-3-Clause)
 #ifdef DESUL_HAVE_HIP_ATOMICS
 #include <desul/atomics/Lock_Based_Fetch_Op_HIP.hpp>
 #endif
-#ifdef DESUL_HAVE_SYCL_ATOMICS
+// FIXME_SYCL Use SYCL_EXT_ONEAPI_DEVICE_GLOBAL when available instead
+#ifdef DESUL_SYCL_DEVICE_GLOBAL_SUPPORTED
```

Comment:
Please confirm that this guy also need to have that special guard, not just the definition of the lock array